sopel/plugins/jobs.py

```diff
@@ -0,0 +1,111 @@
+# coding=utf-8
+"""Sopel's plugin jobs management.
+
+.. versionadded:: 7.1
+
+.. important::
+
+    This is all fresh and new. Its usage and documentation is for Sopel core
+    development and advanced developers. It is subject to rapid changes
+    between versions without much (or any) warning.
+
+    Do **not** build your plugin based on what is here, you do **not** need to.
+
+"""
+# Copyright 2020, Florian Strzelecki <florian.strzelecki@gmail.com>
+#
+# Licensed under the Eiffel Forum License 2.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import itertools
+import logging
+
+from sopel import tools
+from sopel.tools import jobs
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Scheduler(jobs.Scheduler):
+    """Plugins's Job Scheduler
+
+    :param manager: bot instance passed to jobs as argument
+    :type manager: :class:`sopel.bot.Sopel`
+
+    Scheduler that stores plugin jobs and behaves like its
+    :class:`parent class <sopel.tools.jobs.Scheduler>`.
+
+    .. versionadded:: 7.1
+
+    .. note::
+
+        This class is a specific implementation of the scheduler, made to store
+        jobs by their plugins and be used by the bot (its ``manager``).
+        It follows a similar interface as the plugin's
+        :class:`rules manager <sopel.plugins.rules.Manager>`.
```

Comment:
```suggestion
        It follows a similar interface as the
        :class:`plugin rules manager <sopel.plugins.rules.Manager>`.
```

That short-ass line tho, oh well.